fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/DateTimeExtractAndTransform.java

```diff
@@ -494,9 +496,13 @@ public static Expression unixTimestamp(DateV2Literal date) {
         return new IntegerLiteral(getTimestamp(date.toJavaDateType()));
     }
 
-    @ExecFunction(name = "unix_timestamp", argTypes = {"DATETIMEV2"}, returnType = "INT")
+    @ExecFunction(name = "unix_timestamp", argTypes = { "DATETIMEV2" }, returnType = "DECIMALV3")
     public static Expression unixTimestamp(DateTimeV2Literal date) {
-        return new IntegerLiteral(getTimestamp(date.toJavaDateType()));
+        if (date.getMicroSecond() == 0) {
+            return new DecimalV3Literal(new BigDecimal(getTimestamp(date.toJavaDateType()).toString()));
+        }
+        String val = getTimestamp(date.toJavaDateType()).toString() + "." + date.getMicrosecondString();
+        return new DecimalV3Literal(new BigDecimal(val));
```

Comment:
u should create DecimalV3Literal with DecimalV3Type explicitly, because this ctor do not ensure return same precision and scale decimal with original expression. this may lead to wrong result or be crash